x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/planner/TranslatorHandler.java

```diff
@@ -46,7 +46,10 @@ private static Query wrapFunctionQuery(Expression field, Query query) {
         }
         if (field instanceof FieldAttribute fa) {
             fa = fa.getExactInfo().hasExact() ? fa.exactAttribute() : fa;
-            return new SingleValueQuery(query, fa.name(), false);
+            // Extract the real field name from MultiTypeEsField, and use it in the push down query if it is found
+            String fieldNameFromMultiTypeEsField = LucenePushdownPredicates.extractFieldNameFromMultiTypeEsField(fa);
+            String fieldName = fieldNameFromMultiTypeEsField != null ? fieldNameFromMultiTypeEsField : fa.name();
+            return new SingleValueQuery(query, fieldName, false);
```

Comment:
Also here: Separate PR preferred, but the fix is correct (and could be simplified to just always use `fieldName`)